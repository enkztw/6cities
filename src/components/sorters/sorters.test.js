import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

import Sorters from './sorters.jsx';


const mockStore = configureStore([]);


it(`<Sorters /> should have been rendered with Popular`, () => {
  const store = mockStore({
    currentSorting: `id`
  });

  const tree = renderer.create(
      <Provider store={store}>
        <Sorters />
      </Provider>).toJSON();

  expect(tree).toMatchSnapshot();
});
