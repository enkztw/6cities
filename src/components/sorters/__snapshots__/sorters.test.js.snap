// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sorters /> should have been rendered with Popular 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom"
  >
    <li
      className="places__option places__option--active"
      data-name="id"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option"
      data-name="toHigh"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option"
      data-name="toLow"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option"
      data-name="rating"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
